server/src/domain/person/person.service.ts

```diff
@@ -127,22 +145,47 @@ export class PersonService {
     return result;
   }
 
-  async reassignFacesById(auth: AuthDto, personId: string, dto: FaceDto): Promise<PersonResponseDto> {
-    await this.access.requirePermission(auth, Permission.PERSON_WRITE, personId);
-
-    await this.access.requirePermission(auth, Permission.PERSON_CREATE, dto.id);
-    const face = await this.repository.getFaceById(dto.id);
-    const person = await this.findOrFail(personId);
-
-    await this.repository.reassignFace(face.id, personId);
-    if (person.faceAssetId === null) {
-      await this.createNewFeaturePhoto([person.id]);
+  async unassignFace(authUser: AuthDto, id: string): Promise<AssetFaceResponseDto> {
+    let face = await this.repository.getFaceById(id);
+    await this.access.requirePermission(authUser, Permission.PERSON_CREATE, face.id);
+    if (face.personId) {
+      await this.access.requirePermission(authUser, Permission.PERSON_WRITE, face.personId);
```

Comment:
Why do you need these permissions to unassign a face?

Reply:
I always check if the user has access to the face and the person he wants to change. I should maybe create new permissions for that